contrib/upgrade.sh

```diff
@@ -10,6 +10,9 @@
 #   upgrade.sh 1.15.10
 #   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
 
+# load environment variables from `giteaenv` when `giteaenv` exists.
+[[ -f "$(dirname $0)/giteaenv" ]] && source "$(dirname $0)/giteaenv"
```

Comment:
Shouldn't it be responsible for the caller to load the correct env for this script?

Reply:
I just want to set env var in `giteaenv`.

```shell
export giteabin="/opt/gitea/gitea"
export giteahome="/opt/gitea"
export giteaconf="/opt/gitea/app.ini"
export ignore_gpg="yes"
```

Then, call `./upgrade.sh` directly.  This will be very convenient.

Reply:
Problem I have with this approach is that we're hardcoding a specific file `giteaenv` for this purpose.

Reply:
So, what's your suggestion?

Reply:
I also believe the files in `contrib` directory are samples. They are not released with Gitea. Users could just copy and modify these files for specialized purpose. So not very necessary to introduce too many hardcoding.

Reply:
My suggestion is not having this feature, let the user create a wrapper around ./upgrade.sh 🤷🏽.

Reply:
OK